src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -18,89 +18,118 @@
 
 namespace System.Text.RegularExpressions
 {
-    internal sealed class RegexWriter
+    internal ref struct RegexWriter
     {
-        private int[] _intStack;
-        private int _depth;
-        private int[] _emitted;
-        private int _curpos;
-        private readonly Dictionary<string, int> _stringhash;
-        private readonly List<string> _stringtable;
-        private bool _counting;
-        private int _count;
-        private int _trackcount;
-        private Hashtable _caps;
```

Comment:
I think these changes will break someone who used RegexCompileToAssembly on desktop fx trying to then use that assembly in a .NET Core 2.1+ app

Reply:
afaik we don't support that scenario. I was discussing that with Dan previously. And I'm not sure if RegexWriter is cached in the compiled assembly. I don't think so.

Reply:
@joshfree I want to verify, are you sure this breaks the `RegexCompileToAssembly` scenario? I just checked and the only places where the RegexWriter is used in that process is when RegexCode is generated. And only the capsize (int) and the caps (Hashtable) are written into IL from RegexCode.

Reply:
https://referencesource.microsoft.com/#System/regex/system/text/regularexpressions/RegexCompiler.cs,3136

Reply:
sure, will do. I don't think we broke anything here but if we really want to stay compatible with netfx CompileToAssembly generated assemblies we might restrict ourselves in modernizing the existing engine.

Reply:
Ideally we should have a test checked in that runs a compiled regex (emitted by desktop). We have other tests that have small projects to build ancilliary test binaries.

Reply:
if need be, you can persist the test binary asset(s) at https://github.com/dotnet/corefx-testdata